### ISSUE TITLE
[Feature Request] - Progress page

### ISSUE THREAD

**Issue by Ardakilic** [IgnisDa/ryot#996]
Hello,

Trakt.tv has a cool feature to show your watch progress:

https://trakt.tv/users/arda/progress

You could see the overall status of the tv shows that you've been watching from this page. You can see the overall progress, and the lat time when you watched the episode.

This is my one of the most used page of trakt.tv for my account. From this very page, I decide to watch which show's next season to keep on watching.

Is there a page for this in Ryot? If so, how can I achieve this? If not, please consider this as a feature request.

Thanks in advance!

**Comment by IgnisDa** [comment]
Thanks. I will consider this. Is there anything similar for books?

**Comment by Ardakilic** [comment]
@IgnisDa thanks, not that I know of. This page is the page I visit on Trakt most, to decide what to watch next, and overall progress page could be helpful to see what I've watched so far.


For movies and TV shows (also mixed) lists could be utilized, [just like this](https://trakt.tv/users/username57/lists/mcu-timeline-marvel-studios?sort=rank,asc), which could be mixed with TV show episodes [like this](https://trakt.tv/users/username57/lists/star-trek-prime-canon-timeline?sort=rank,asc), and the desired state could be tracked within the list, but that's something totally different as a technical scope. Someone has to create and keep the list up-to-date in the first place. Apart from this there's not a thing that I know of for movies exists.

**Comment by arpanghosh8453** [comment]
Here is a sample for the books : 

![image](https://github.com/user-attachments/assets/4264351d-f257-480b-9f99-24610a69ddcd)

**Comment by Ardakilic** [comment]
Something like that would be cool, but not enough for my initial post requirement. I think it holds the state of a single book. You don't usually switch to other books mid-chapter.

That's why it'd be great if it'd show the "last watched date" (and sorted by this date, descending) and "next episode" so we could decide what to pick to watch next from the progress page.

"Next episode" information is nice to have, because at a given time the new season may not be aired yet but later, or the show would be completed altogether so we could decide what to pick without digging into each show's details.

**Comment by IgnisDa** [comment]
@arpanghosh8453 Could you create a new issue for this? @Ardakilic is right, this is a different issue. Sorry for sending you here 😅.

**Comment by IgnisDa** [comment]
@arpanghosh8453 The feature you requested for has now been merged. Will be released this weekend.

**Comment by arpanghosh8453** [comment]
@IgnisDa Thank you so much! Appreciate it!

**Comment by olehkrupko** [comment]
Hello! Just started setting up Ryot as data backup solution after long years of using Trakt and its Progress page. Maybe I'll be able to shed some light on how it works for me compared to first impressions of Ryot.

# TLDR
1. Instead of manually adding stuff you watch to separate Collection, Progress page displays all shows you ever started except ones you excluded from there. It's useful to make sure you finished all shows you ever started watching. Ryot search filters have "Dropped" in first dropdown, no idea where to find it elsewhere, but it might be a good alternative to Trakt's hidden items.
2. Progress view has lots of sorting/filtering options above with the main feature of hiding any of the following (multiple options can be chosen): Completed, Not Completed, Ended / Canceled, Currently Airing, Rewatching. I'm constantly hiding Completed shows and sorting by last watched.
3. There is no similar interface for movies and having one will be awesome.

# How it works right now

## Ryot

I like Ryot. I want to say it beforehand as I'll make some comparisons and I don't really understand how Ryot is supposed to work. I want to have a similar experience to Progress page and if it's not possible or Ryot itself works in a different way that I want, then it's alright: everyone has different tastes and I respect amount of work you did. Will write too much text as you might find something useful. Just in case: I created my instance on V7, imported Trakt data and upgraded it to V8 later.

<img width="962" alt="Image" src="https://github.com/user-attachments/assets/3afb8724-6296-481e-96e3-ad9a5ec97d6a" />

- In Progress section of Dashboard displays items from "In Progress" Collection.
- I discovered it after some experimentation and it will be nice if it was clearly mentioned in Collection's description.
- For me, it seems to be same thing as "Monitoring" Collection, maybe it's for Arr service integration if there is one, but it's not point of discussion there.
- Card are used as display style, with three icons: "Finished", "Play" (Marks as watched) and "Star" (leave a rating).

<img width="507" alt="Image" src="https://github.com/user-attachments/assets/e7b09ed6-5f3a-42f2-8195-753f8d6a30ed" />

- Clicking "Play" shows form above. You need to click "Submit" to finish submitting your progress.

## Trakt
Instead of manually adding stuff you watch to separate Collection, Progress page displays all shows you ever started watching except ones you excluded from there. It's useful to make sure you finished all shows you ever started watching. Hidden shows have a separate page in setting that can be used to unhide them.

There are two views for show progress: cards and detailed (not official name, it's the one above)

### Cards
Screenshot taken from Trakt Dashboard, Progress page can show cards as well (cards same, interface around is different).

<img width="1174" alt="Image" src="https://github.com/user-attachments/assets/81a910eb-6ef1-4fae-b61f-aa1958b7b6d4" />

- Cards are bigger, they are approximately 50% higher allowing to squeeze more stuff in.
- At the top, there is description bar, examples might be "Season Premiere" or "Series Finale", not sure in full list of possible text.
- At the bottom we have "Checkmark" (mark as watched), "Book" (I believe it's an alternative to "Owned" Collection on Ryot), "List" (adds to Watchlist by default or shows checklist of all lists you have if you created additional lists), "Play" (shows where you can stream, rent or purchase it) and "Rating" (shows global rating and clicking allows to rate item). "Book" and "List" buttons can have blue background for cases when item is collected or added to Watchlist.
- Global show progress bar. Looks cool, I love it.
- Episode number and name.
- Show name with a few buttons: "Rewind" (don't use, but useful for people rewatching shows) and "Hide" (hide from Progress page)

### Filters and detailed

<img width="1161" alt="Image" src="https://github.com/user-attachments/assets/caa3700e-529f-4cc1-9f2b-8ab37edb567a" />

1. Watched/Collected dropdown, I guess you can follow progress of collected items as well.
2. Progress numbers: percents, time to watch and amount of shows.
3. Sorting options: Watched Date, Completion %, Episodes Left, Time Left, Plays, Release Date, Premiere Date, Title, Popularity, Episode Runtime, Total Runtime and Random. I like "Watched Date": recently watched shows are displayed first.
4. Card view toggle.
5. Advanced Filtering (Trakt VIP required) for shows, I believe these ones: by watch now availability, genre, certification, language, country, network, year, runtime and rating.
6. Filter by streaming service (Trakt VIP required).
7. Progress filters. That's an important one. Has list of items with checkmarks. You choose which shows to hide from Progress page: Completed, Not Completed, Ended / Canceled, Currently Airing, Rewatching.
8. RSS Feed (Trakt VIP required).
9. CSV Export (Trakt VIP required).

# Conclusions, ideas and feedback

1. **Smart Collections**. Instead of manually adding stuff you watch to a separate Collection, I would like to see a "smart" Collection (I believe photo management software calls it a Smart Album and file explorers call it smart folders) as it will allow us to configure what is shown in "In Progress" Dashboard widget. In case of "In Progress" Collection, it will include all shows which are not marked as Dropped. As an alternative, we will be able to create "smart" collections from any Ryot search search query and configure which shows we would like to see on "In Progress" collection this way. Ability to create more smart collections can be a payed feature, but please, leave some default free space if you make it Pro.
2. **System-created Collections**:
    - Clearer indication that "In Progress" Collection is used for "In Progress" dashboard widget. It will be nice if this Collection was linked from Settings >> Preferences >> Dashboard >> In Progress and its usage was described in Collection description.
    - What is "Monitoring" Collection? By name, it seems related to Progress topic, but I'm not sure how it's different from "In Progress" and what it's used for except for "Start monitoring" button in show actions.
4. **Search Filters**. Ryot search filters have "Dropped" option in first dropdown, no idea where to find it elsewhere as show actions don't seem to have it, but it might be a good alternative to Trakt's hidden items. Is it different from "On Hold" filter?
5. **Docs**: I guess it will be nice to have at least some text in docs explaining how to use search and collections. Maybe I missed it at https://docs.ryot.io/ ?
6. **Submitting show progress**. It will be nice if immediately clicking "Enter" on keyboard when "Play" form is shown submitted this form. I did not expect "Play" to be about marking progress (I expected a checkmark) thinking it will lead to "Watch On" tab of show description.

Let me know if you would like to discuss anything.

**Comment by aksdb** [comment]
I haven't worked with trakt, so I don't know how they handle rewatches, but what I miss from all the other tools/services I used so far (watcharr, ryot, justwatch, emby, etc.), the ability to track rewatches independently from previous watches. 

Naively I could imagine a solution could be to track progress of a tv show as part of the list they are in. Then I can put a show I already watched in a new "rewatch 2025"-list (as an example), where its progress would be 0 and I can keep track of the episodes I watched so far, without losing the progress of my first, second, etc. (re)watches years ago. The amount (and maybe even times) of rewatches could still be tracked _additionally_ on the base entry for an episode (so I can see accumulated how often I watched something and maybe even when), but the progress would be nice to have decoupled (IMO).

**Comment by IgnisDa** [comment]
What do you think about views in the list page? Somewhat similar to how https://plane.so does it

<img width="326" height="102" alt="Image" src="https://github.com/user-attachments/assets/7b494934-f18a-49df-a0d1-5bf208ce5489" />

**Comment by UptightGlitter** [comment]
This is something I miss a lot and would be a great QoL feature!

A similar section to highlight unwatched episodes of ongoing series and new premieres  would allow users to quickly see which episodes they've missed and stay up-to-date with their shows.

Here is what I envisioned:

- **Episode Cards:**  
  Each show appears as a card featuring the series artwork, with a badge highlighting the number of unwatched episodes.
- **Clear Episode Labels:**  
  Beneath each card, the next unwatched episode is labeled (e.g., "S01 E02" or "Ep. 7") and the total number of episodes.
- **Sorting:**  
  Shows are ordered by the last aired episode with unwatched content, ensuring new releases and recently updated series are prioritized.

This would enables quick access to what's new and pending for the user, promoting a streamlined and visually engaging way to keep track of progress.

**Comment by Ardakilic** [comment]
Hello @IgnisDa , just wondering if there's any plans for this in the near future. I believe this is an essential overview page for a TV show viewer if you're jumping from a show to other. Thanks!

**Comment by IgnisDa** [comment]
I have something in the works which will easily address this but
unfortunately it's not in a state that it can be revealed in it's current
state.

I'll drop back here soon though!

**Comment by Ardakilic** [comment]
@IgnisDa thank you! waiting for the good news! This now way more important for me, since Trakt revamped their whole website and they accidentally killed this progress page.
